kotlin-source/src/main/kotlin/com/example/api/ExampleApi.kt

```diff
@@ -71,23 +75,25 @@ class ExampleApi(val services: CordaRPCOps) {
                 otherParty,
                 IOUContract())
 
-        // The line below blocks and waits for the future to resolve.
-        val result: ExampleFlowResult = services
-                .startFlow(::Initiator, state, otherParty)
-                .returnValue
-                .getOrThrow()
+        val (status, msg) = try {
+            // The line below blocks and waits for the future to resolve.
+            val result = services
+                    .startFlow(::Initiator, state, otherParty)
+                    .returnValue
+                    .getOrThrow()
 
-        when (result) {
-            is ExampleFlowResult.Success ->
-                return Response
-                        .status(Response.Status.CREATED)
-                        .entity(result.message)
-                        .build()
-            is ExampleFlowResult.Failure ->
-                return Response
-                        .status(Response.Status.BAD_REQUEST)
-                        .entity(result.message)
-                        .build()
+            Response.Status.CREATED to "Transaction id ${result.id} committed to ledger."
+
+        } catch (ex: Throwable) {
+            val msg = when (ex) {
```

Comment:
This catch block isn't right. You can need to be looking at the ExecutionException cause